[PATCH 1/1] ERL: fix relocation of common symbols

The current ERL library fails to relocate global common symbols (placed in the
COMMON and .scommon sections). With this patch, the linker is told to assign
space to common symbols, thereby moving them to the .bss section.

Reported-by: jimmikaelkael <jimmikaelkael@wanadoo.fr>
Signed-off-by: misfire <misfire@xploderfreax.de>

diff --git a/ee/Rules.make b/ee/Rules.make
--- a/ee/Rules.make
+++ b/ee/Rules.make
@@ -58,5 +58,5 @@
 	$(EE_AR) cru $(EE_LIB) $(EE_OBJS)
 
 $(EE_LIB:%.a=%.erl) : $(EE_OBJS)
-	$(EE_CC) -mno-crt0 -Wl,-r -o $(EE_LIB:%.a=%.erl) $(EE_OBJS)
+	$(EE_CC) -mno-crt0 -Wl,-r -Wl,-d -o $(EE_LIB:%.a=%.erl) $(EE_OBJS)
 	$(EE_STRIP) --strip-unneeded -R .mdebug.eabi64 -R .reginfo -R .comment $(EE_LIB:%.a=%.erl)
diff --git a/samples/Makefile.eeglobal_sample b/samples/Makefile.eeglobal_sample
--- a/samples/Makefile.eeglobal_sample
+++ b/samples/Makefile.eeglobal_sample
@@ -54,7 +54,7 @@
 		-o $(EE_BIN) $(PS2SDK)/ee/startup/crt0.o $(EE_OBJS) $(EE_LDFLAGS) $(EE_LIBS)
 
 $(EE_ERL) : $(EE_OBJS)
-	$(EE_CC) -mno-crt0 -o $(EE_ERL) $(EE_OBJS) $(EE_CFLAGS) $(EE_LDFLAGS) -Wl,-r
+	$(EE_CC) -mno-crt0 -o $(EE_ERL) $(EE_OBJS) $(EE_CFLAGS) $(EE_LDFLAGS) -Wl,-r -Wl,-d
 	$(EE_STRIP) --strip-unneeded -R .mdebug.eabi64 -R .reginfo -R .comment $(EE_ERL)
 
 $(EE_LIB) : $(EE_OBJS)
